From b51f5d8c4fab9c2ba19592a390e32d95507da9be Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Sun, 21 Feb 2016 12:38:45 +0200
Subject: [PATCH 137/538] cp110: ahci: added support for ahci unit

Added support for a dual port AHCI unit for cp110.
Enabled the sata port for all 7040 RZ board setups

Change-Id: I7f2c903e8db0e659f7030f69e17f314f7e912fd2
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-rz-db-nand.dts   |  7 +++++++
 arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts |  7 +++++++
 arch/arm64/boot/dts/marvell/armada-7040-rz-db-rxaui.dts  |  7 +++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi            | 10 ++++++++++
 4 files changed, 31 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-nand.dts
index 14eb472..544b303 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-nand.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-nand.dts
@@ -55,4 +55,11 @@
 
 	chosen { };
 
+	cpn-110-master {
+		config-space {
+			sata: sata@540000 {
+				status = "okay";
+			};
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts
index 0cb4922..74e75d0 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts
@@ -55,4 +55,11 @@
 
 	chosen { };
 
+	cpn-110-master {
+		config-space {
+			sata: sata@540000 {
+				status = "okay";
+			};
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-rxaui.dts b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-rxaui.dts
index 3444c71..f09bb35 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-rxaui.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-rxaui.dts
@@ -55,4 +55,11 @@
 
 	chosen { };
 
+	cpn-110-master {
+		config-space {
+			sata: sata@540000 {
+				status = "okay";
+			};
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 289c6ed..12acaf7 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -29,3 +29,13 @@ gateclk: clock-gating-control@440220 {
 	clocks = <&cp110_clk 3>;
 	#clock-cells = <1>;
 };
+
+sata: sata@540000 {
+	compatible = "marvell,armada-3700-ahci";
+	reg = <0x540000 0x30000>;
+	interrupts = <GIC_SPI 63 IRQ_TYPE_LEVEL_HIGH>;
+	clocks = <&gateclk 15>, <&gateclk 16>;
+	status = "disabled";
+	port_base = <0x10000>;
+	port_offset = <0x10000>;
+};
-- 
1.9.1

